### PR TITLE
[CDF-333] - API documentation for Logger

### PR DESCRIPTION
**Pull request by pamval**

### CODE REVIEW

**Comment by pamval** [comment]
@jvelasques @afrjorge @diogofscmariano please review

File: cdf-core/cdf/js-modules/dashboard/Dashboard.lifecycle.js [hunk 1]
```diff
@@ -78,8 +93,11 @@ define(['./Dashboard', '../Logger', 'amd!../lib/underscore', '../components/Unma
     },
 
     /**
+     * Init function for the dashboard. Calling this method will trigger the dashboard execution and render
      *
-     * @param components
+     * @param components - Components to be added to the dashboard
+     * @method  decrementRunningCalls
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/dashboard/Dashboard.lifecycle.js]: method init

File: cdf-core/cdf/js-modules/dashboard/Dashboard.lifecycle.js [hunk 2]
```diff
@@ -217,17 +239,17 @@ define(['./Dashboard', '../Logger', 'amd!../lib/underscore', '../components/Unma
     },
 
     /**
+     * HAndles the postInit section of the dashboard initialization
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/dashboard/Dashboard.lifecycle.js]: Upper case 'A'  should be lower case

File: cdf-core/cdf/js-modules/dashboard/Dashboard.lifecycle.js [hunk 3]
```diff
@@ -506,8 +534,12 @@ define(['./Dashboard', '../Logger', 'amd!../lib/underscore', '../components/Unma
     },
 
     /**
+     * Updates a specific component
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/dashboard/Dashboard.lifecycle.js]: Is this correct? I got the idea that this method gathers components to update (as the comment inside the method describes), and then calls updateAll with all the gathered components.
UpdateAll in turn readies a queue for updating, and then calls updateComponent on each component, this final one being the actual method which updates a specific component

File: cdf-core/cdf/js-modules/dashboard/Dashboard.legacy.js [hunk 4]
```diff
@@ -176,7 +271,18 @@ define(['../queries/CdaQuery.ext', '../components/XactionComponent.ext', './Dash
       };
       $.post(DashboardExt.getSettings("set", null), data, function(){});
     },
-    
+
+
+    /**
+     * gets a setting value from the server
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/dashboard/Dashboard.legacy.js]: Uppercase 'g'

File: cdf-core/cdf/js-modules/dashboard/Dashboard.legacy.js [hunk 5]
```diff
@@ -185,7 +291,17 @@ define(['../queries/CdaQuery.ext', '../components/XactionComponent.ext', './Dash
     
       $.getJSON(DashboardExt.getSettings("get", key), callback);
     },
-    
+
+    /**
+     * Fetches data from the server according to a ChartDefinition object.
+     * This method is deprecated. {{#crossLink "Query"}}Query{{/crossLink}} object should be used instead
+     * @param cd Chart Definition object
```
Comment (jvelasques) [inline cdf-core/cdf/js-modules/dashboard/Dashboard.legacy.js]: missing method tag

**Pull request closed without merging**